capstone/src/main/java/com/google/sps/servlets/PollServlet.java

```diff
@@ -16,25 +16,32 @@
 import com.google.appengine.api.datastore.Query;
 import com.google.gson.Gson;
 import com.google.sps.Objects.Option;
+import com.google.sps.Objects.response.PollResponse;
 
 @WebServlet("/poll")
 public class PollServlet extends HttpServlet {
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String userId = "200";
     Query query = new Query("Option");
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery results = datastore.prepare(query);
     List<Option> options = new ArrayList<Option>();
+    List<Long> votedOptions = new ArrayList<Long>();
     for (Entity entity : results.asIterable()) {
       long id = entity.getKey().getId();
       String text = (String) entity.getProperty("text");
       List<String> votes = (ArrayList<String>) entity.getProperty("votes");
       Option option = new Option(id, text, votes);
       options.add(option);
+      // If current user voted for option, add to list of voted options
+      if (votes != null && votes.contains(userId)) {
+        votedOptions.add(id);
+      }
```

Comment:
If I am only checking once whether votes.contains(userId) that is total of O(n) for the entire method. Converting it to a HashSet would also be O(n), so why would making it a HashSet be better? I am not going to be using this HashSet later on. It is only created and used within this method once

Reply:
Thank you for being patient and I apologize for taking a while to respond, I've been thinking about this problem the entire day. 
Firstly regarding the runtime of this method, while you are right that converting to a HashSet would be O(n), since you need to iterate through each of the Option Entities, the runtime for the method as a whole would be more than O(n). However whether we convert votes to a HashSet or use the contains list method, both are O(n) complexity, therefore there isn't an apparent initial advantage to using one or the other. 
So, assuming your point about "it is only created and used within this method once" is correct and based on your comment in PollResponse above, then I think it is fine to leave it as is.